natsstreaming/nats_streaming.go

```diff
@@ -212,18 +249,21 @@ func (c *asyncMessageSource) ConsumeMessages(ctx context.Context, messages chan<
 			return
 		case messages <- cm:
 		}
+		if rehydration && msg.Sequence == latestOffset {
+			rehydrationCh <- struct{}{}
```

Comment:
this can cause a deadlock on context cancellation

Reply:
A deadlock on client-side you mean? Not really as I can pass a context in `WithRehydration`.

So, based on my example, I'd do that:

```go
select {
case <-timeout.Done():
	// Error
case <-ch:
	// Set ready
}
```

Reply:
No I mean if the client is not listening on this channel (only if it's not buffered) then the call to consume will get stuck here forever...

Reply:
The goal of this mechanism is to wait for a notification before to set the application ready. If the client is not listening to the channel then he's an idiot.

What else do you propose?